#!/usr/bin/env python
PACKAGE = "collision_avoidance"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

#           Name                      Type       Level  Description                                     Default       Min     Max
gen.add("height",                     double_t,    1,   "Height of the robot (m)",                      0.5,          0,      5)

gen.add("radius",                     double_t,    2,   "Radius of the robot (m)",                      0.60,         0,      5)
gen.add("security_distance",          double_t,    2,   "Security distance (m)",                        1.0,          0,      2)
gen.add("epsilon",                    double_t,    2,   "Epsilon in ORM",                               0.1,          0,      1)
gen.add("polar_size",                 int_t,       2,   "polar_size in ORMT",                           360,          0,      3600)

gen.add("min_distance_hold",          double_t,    3,   "Obstacles considered when no input (m)",       4.0,          0,      10)

gen.add("h_m",                        double_t,    4,   "Emperically measured constant for speed",      2.0,          0.01,   10)
gen.add("max_xy_vel",                 double_t,    4,   "Maximum horizontal velocity (m/s)",            0.25,         0,      10)
gen.add("max_z_vel",                  double_t,    4,   "Maximum vertical velocity (m/s)",              0.25,         0,      10)
gen.add("max_yaw_rate",               double_t,    4,   "Maximum yaw_rate velocity (deg/s)",            10,           0,      360)

gen.add("distance_converged",         double_t,    5,   "Distance converged (m)",                       0.1,          0,      1)
gen.add("yaw_converged",              double_t,    5,   "Yaw converged (rad)",                          0.1,          0,      1)
gen.add("frequency",                  double_t,    5,   "Frequency",                                    20,           1,      100)

gen.add("max_times_backwards",        int_t,       6,   "Max times backwards in a row",                 200,          0,      10000)
gen.add("max_direction_change",       double_t,    6,   "Max direction change",                         45,           0,      180)

gen.add("leaf_size",                  double_t,    7,   "Leaf size",                                    0.05,         0,      0.2)

gen.add("look_ahead_distance",        double_t,    8,   "Look ahead distance",                          4.0,          0,      10)
gen.add("look_forward",               bool_t,      8,   "Look forward",                               True)
gen.add("move_while_yawing",          bool_t,      8,   "Move while yawing",                             False)
gen.add("yaw_each_setpoint",           bool_t,      8,   "Yaw each setpoint",                            False)

exit(gen.generate(PACKAGE, "collision_avoidance", "CollisionAvoidance"))
